src/state/apis/zapper/zapperApi.ts

```diff
@@ -89,7 +114,58 @@ export const zapperApi = createApi({
         return { data }
       },
     }),
+    getZapperNftUserTokens: build.query<V2NftUserItem[], GetZapperNftUserTokens>({
+      queryFn: async ({ accountIds }) => {
+        let data: V2NftUserItem[] = []
+
+        const userAddresses = accountIdsToEvmAddresses(accountIds)
+
+        for (const userAddress of userAddresses) {
+          // https://studio.zapper.fi/docs/apis/api-syntax#v2nftusertokens
+          /**
+           * docs about cursor are wrong lmeow
+           * check the length of returned items to see if there are more
+           */
+          const limit = 100
+          while (true) {
+            try {
+              const res = await zapperClient.getV2NftUserTokens({
+                // Encode query params with arrayFormat: 'repeat' because zapper api derpexcts it
+                paramsSerializer: params => qs.stringify(params, { arrayFormat: 'repeat' }),
+                headers,
+                queries: { userAddress, limit: limit.toString() },
+              })
+              if (!res?.items?.length) break
+              if (res?.items?.length) data = data.concat(res.items)
+              if (res?.items?.length < limit) break
+            } catch (e) {
+              moduleLogger.warn(e, 'getZapperNftUserTokens')
+              break
+            }
+          }
+        }
+        return { data }
+      },
+    }),
+    getZapperCollections: build.query<V2NftCollectionType[], GetZapperCollections>({
+      queryFn: async ({ accountIds, collectionAddresses }) => {
+        const addresses = accountIdsToEvmAddresses(accountIds)
+        const { items: data } = await zapperClient.getV2NftBalancesCollections({
+          headers,
+          // yes this is actually how the api expects it
```

Comment:
akschually either or, but this works better with zodios typings and params serialization